src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/src/utils/generate_static_shapes.cpp

```diff
@@ -0,0 +1,137 @@
+#include "shared_test_classes/single_op/roi_align.hpp"
+
+#include "common_test_utils/data_utils.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+uint64_t clip(uint64_t n, uint64_t lower, uint64_t upper) {
+    return std::max(lower, std::min(n, upper));
+}
+
+void clip_restrict_dims(InputShape& input_shape, const std::map<size_t, size_t>& restrict_dims) {
+    std::vector<ov::Shape>& staticShapes = input_shape.second;
+    for (const auto& pair : restrict_dims) {
+        uint64_t dimMax = std::numeric_limits<char>::max();
+        uint64_t dimMin = pair.second;
+        auto& dim0 = staticShapes[0][pair.first];
+        auto& dim1 = staticShapes[1][pair.first];
+        auto& dim2 = staticShapes[2][pair.first];
+        if (dim0 != dim2) {
+            dim0 = clip(dim0, dimMin, dimMax);
+            dim1 = clip(dim1, dimMin, dimMax);
+            dim2 = clip(dim2, dimMin, dimMax);
+        }
+    }
+}
+
+namespace {
+
+InputShape generate(const std::shared_ptr<ov::Node>& node,
+                    size_t in_port_id) {
+    const auto& param = std::dynamic_pointer_cast<ov::op::v0::Parameter>(node->get_input_node_shared_ptr(in_port_id));
+    std::vector<ov::Shape> staticShapes = { param->get_partial_shape().get_min_shape(),
+                                            param->get_partial_shape().get_min_shape(),
+                                            param->get_partial_shape().get_max_shape() };
+    ov::Shape midShape;
+    for (const auto s : param->get_partial_shape()) {
+        int dimValue = 1;
+        if (s.is_dynamic()) {
+            size_t range = s.get_max_length() - s.get_min_length();
+            if (range > std::numeric_limits<char>::max()) {
+                ov::test::utils::fill_data_random(&range, 1, std::numeric_limits<char>::max(), s.get_min_length(), 1);
+            } else {
+                ov::test::utils::fill_data_random(&dimValue, 1, range, s.get_min_length(), 1);
+            }
+        } else {
+            dimValue = s.get_length();
+        }
+        midShape.push_back(dimValue);
+    }
+    staticShapes[1] = midShape;
+
+    // Shape validation to avoid large values
+    uint64_t dimMin = 1;
+    uint64_t dimMax = std::numeric_limits<char>::max();
+    for (int i = 0; i < staticShapes[0].size(); ++i) {
+        auto& dim0 = staticShapes[0][i];
+        auto& dim2 = staticShapes[2][i];
+        if (dim0 != dim2) {
+            dim0 = clip(dim0, dimMin, dimMax);
+            dim2 = clip(dim2, dimMin, dimMax);
+        }
+    }
+    return InputShape{param->get_partial_shape(), staticShapes};
+}
+
+InputShape generate(const std::shared_ptr<ov::op::v1::Convolution>& node,
+                    size_t in_port_id) {
+    std::map<size_t, size_t> restrict_dims;
+    for (size_t port = 0; port < node->get_input_size(); ++port) {
+        if (port != in_port_id) {
+            auto constant_ptr = std::dynamic_pointer_cast<ov::op::v0::Constant>(node->get_input_node_shared_ptr(port));
```

Comment:
Because the implementation is specific to `Convolution`, the process unfolds as follows:
1. Initially, we generate static shapes for the **non-constant input** (port == in_port_id).

2. Subsequently, we clip the static shapes based on the last dimension of the op attribute Kernel (constant input of which port != in_port_id).

I believe the `kernel` in the `Convolution` operation should always remain constant ? If there's a necessity to restrict the input of the `Convolution` operation based on other attributes, the logic should be adjusted accordingly?